mc-sema/cfgToLLVM/linuxArchOps.cpp

```diff
@@ -249,78 +247,82 @@ Value *linuxFreeStack(Module *M, Value *stackAlloc, BasicBlock *&driverBB) {
 
 llvm::Value *linuxMakeCallbackForLocalFunction(Module *M , VA local_target) {
 
-    if(getSystemArch(M) == _X86_64_) {
-        std::string	call_tgt_name = "sub_" + to_string<VA>(local_target, std::hex);
-        std::string	callback_name = "callback_sub_" + to_string<VA>(local_target, std::hex);
+    std::string	call_tgt_name = "sub_" + to_string<VA>(local_target, std::hex);
+    std::string	callback_name = "callback_sub_" + to_string<VA>(local_target, std::hex);
 
-        Function *call_tgt = M->getFunction(call_tgt_name);
-        TASSERT( call_tgt != NULL, "Cannot find target for callback");
+    Function *call_tgt = M->getFunction(call_tgt_name);
+    TASSERT( call_tgt != NULL, "Cannot find target for callback");
 
-        Function *callback = M->getFunction(callback_name);
+    Function *callback = M->getFunction(callback_name);
 
-        // already have a callback
-        if(callback != nullptr) {
-            return callback;
-        }
+    // already have a callback
+    if(callback != nullptr) {
+        return callback;
+    }
 
-        unsigned regWidth = getPointerSize(M);
+    unsigned regWidth = getPointerSize(M);
 
-        // no driver, make one
-        FunctionType    *callbackTy = 
-            FunctionType::get( Type::getIntNTy(M->getContext(), regWidth), false );
+    // no driver, make one
+    FunctionType    *callbackTy = 
+        FunctionType::get( Type::getIntNTy(M->getContext(), regWidth), false );
 
-        Function *F = dynamic_cast<Function*>(M->getOrInsertFunction(callback_name, callbackTy));
-        TASSERT( F != NULL, "Cannot create callback stub" );
-        F->setLinkage(GlobalValue::InternalLinkage);
-        F->addFnAttr(Attribute::Naked);
+    Function *F = dynamic_cast<Function*>(M->getOrInsertFunction(callback_name, callbackTy));
+    TASSERT( F != NULL, "Cannot create callback stub" );
+    F->setLinkage(GlobalValue::InternalLinkage);
+    F->addFnAttr(Attribute::Naked);
 
-        // add code to driver
-        BasicBlock *block = BasicBlock::Create(
-                F->getContext(), "", F);
+    // add code to driver
+    BasicBlock *block = BasicBlock::Create(
+            F->getContext(), "", F);
 
-        std::vector<Type*>FuncTy_9_args;
-        FuncTy_9_args.push_back(call_tgt->getType());
-        FunctionType* FuncTy_9 = FunctionType::get(
-                /*Result=*/Type::getVoidTy(M->getContext()),
-                /*Params=*/FuncTy_9_args,
-                /*isVarArg=*/false);
+    std::vector<Type*>FuncTy_9_args;
+    FuncTy_9_args.push_back(call_tgt->getType());
+    FunctionType* FuncTy_9 = FunctionType::get(
+            /*Result=*/Type::getVoidTy(M->getContext()),
+            /*Params=*/FuncTy_9_args,
+            /*isVarArg=*/false);
+
+    if(getSystemArch(M) == _X86_64_) {
 
         InlineAsm* ptr_13 = InlineAsm::get(FuncTy_9, "movq $0, %rax\n", "imr,~{dirflag},~{fpsr},~{flags}",true);
```

Comment:
Ahh so this is LLVM inline asm. `$0` is not a literal `0`, but instead a reference to the first operand.